alf/utils/value_ops.py

```diff
@@ -66,7 +66,14 @@ def action_importance_ratio(action_distribution, collect_action_distribution,
 
     action_log_prob = dist_utils.compute_log_probability(
         current_policy_distribution, action)
+
     if log_prob_clipping > 0.0:
+        # here we need to clamp the sample_action_log_probs as it is computed
+        # using actions that not sampled from the collect_action_distribution
+        # itself but from another distribution and is more prone to have
+        # numerical issues
+        sample_action_log_probs = sample_action_log_probs.clamp(
+            -log_prob_clipping, log_prob_clipping)
```

Comment:
I think in this case we should just *turn off* cache and force the distribution to compute the inverse, e.g., using Beta with cache_size=0?

Reply:
Cache does not play a role in the computation in this case, and that's the reason for unstable computation.
I have already using beta which seems to be more stable.

Reply:
> Cache does not play a role in the computation in this case, and that's the reason for unstable computation. I have already using beta which seems to be more stable.

Right. If the 'y' object is not the same with the cached one, then it will compute.

Reply:
> Cache does not play a role in the computation in this case, and that's the reason for unstable computation. I have already using beta which seems to be more stable.

You could try BetaProjectionNetwork with min_concentration=1. or Beta with eps=1e-5.